_posts/2015-03-31-flagon.markdown

```diff
@@ -0,0 +1,132 @@
+
+Having said that, I'm not going to waggle my finger at you and tell you you're doing it wrong. If you want to use the database for feature flags in some cases, so be it! You can. Flagon allows any storage mechanism if you care to extend it.
+
+## Using Flagon, the simplest example
+
+In it's simplest form, Flagon can be set up to use environment variables in one line:
+
+{% highlight ruby %}
+Flagon.init
+{% endhighlight %}
+
+Flagon expects environment variables to be set to "on". If they are not "on" then they are considered off. So to set up your first flag you could set an environment variable like so:
+
+{% highlight bash %}
+export DAILY_EMAILS=on
+{% endhighlight %}
+
+Then, you would check the flag in your project by calling:
+
+{% highlight ruby %}
+Flagon.enabled?(:daily_emails)
+{% endhighlight %}
+
+## Alternative ways of defining flags
+
+You might not want to use environment variables all the time, maybe you want something even simpler, like a hash? You can do this by initializing a hash loader and passing it into the init method of Flagon:
+
+{% highlight ruby %}
+loader = Flagon.hash_loader { daily_emails: "on" }
+Flagon.init loader
+{% endhighlight %}
+
+Other than that change (and not needing to set environment variables), you use Flagon in exactly the same way as before. To keep things consistent with the environment variables, we stuck with "on" to define an enabled flag.
+
+In addition to the hash loader, Flagon also provides the file loader which will read from a yaml file:
+
+{% highlight ruby %}
+loader = Flagon.file_loader "/path/to/my/file.yaml"
+Flagon.init loader
+{% endhighlight %}
+
+The yaml file is simple too:
+
+{% highlight ruby %}
+daily_emails: on
+automatic_happy_face_emoji: off
+{% endhighlight %}
+
+## Extending Flagon
+
+If environment variables, hashes and yaml aren't your thing you can create your own loaders too. You just need to implement the loader interface on a custom object and pass it in instead of one of the provided loaders. Here's the loader interface:
+
+{% highlight ruby %}
+# returns true or false, depending on whether the flag has been defined.
+exists?(:flag_name)
+
+# returns true or false, depending on whether the flag is "on".
+get_flag(:flag_name)
+{% endhighlight %}
+
+Flagon will call those two methods to work out whether a flag is on or off and your application can be blissfully ignorant about how or where the flags are defined in development, test or production.
+
+For example, you could create an active record model with two attributes "name" and "value" and implement the two interface methods like so:
+
+{% highlight ruby %}
+class Flag < ::ActiveRecord::Base
+  def self.exists?(flag_name)
+    where(name: flag_name).count > 0
```

Comment:
is there a better way to do it that isn't exists? E.g. `where(name: flag_name).first.present?`